engine/crates/integration-tests/src/federation/builder/test_runtime.rs

```diff
@@ -18,11 +18,14 @@ pub struct TestRuntime {
     pub hooks: DynamicHooks,
     pub rate_limiter: runtime::rate_limiting::RateLimiter,
     pub entity_cache: InMemoryEntityCache,
+    pub access_log_sender: runtime_local::hooks::ChannelLogSender,
+    pub access_log_receiver: runtime_local::hooks::ChannelLogReceiver,
 }
 
 impl Default for TestRuntime {
     fn default() -> Self {
         let (_, rx) = watch::channel(Default::default());
+        let (access_log_sender, access_log_receiver) = runtime_local::hooks::create_log_channel();
```

Comment:
I don't think I would put it here, `TestRuntime` will be owned by Engine. So you lose the receiver, the receiver should be added to `TestEngineV2` instead so it's accessible inside tests.